move/squid/sources/squid/squid.move

```diff
@@ -1,36 +1,69 @@
 module squid::squid;
 
+use std::ascii;
+
+use sui::clock::Clock;
+use sui::versioned::{Self, Versioned};
+
+use version_control::version_control::{Self, VersionControl};
+
 use axelar_gateway::channel::{Self, Channel, ApprovedMessage};
+
 use its::its::ITS;
 use its::service;
-use squid::coin_bag::{Self, CoinBag};
-use squid::swap_info::{Self, SwapInfo};
-use sui::clock::Clock;
-use version_control::version_control::{Self, VersionControl};
 
+use squid::coin_bag::CoinBag;
+use squid::swap_info::SwapInfo;
+use squid::squid_v0::{Self, SquidV0};
 // -------
 // Version
 // -------
 const VERSION: u64 = 0;
 
 public struct Squid has key, store {
     id: UID,
-    channel: Channel,
-    coin_bag: CoinBag,
-    version_control: VersionControl,
+    inner: Versioned,
 }
 
 fun init(ctx: &mut TxContext) {
     transfer::share_object(Squid {
         id: object::new(ctx),
-        channel: channel::new(ctx),
-        coin_bag: coin_bag::new(ctx),
-        version_control: new_version_control(),
+        inner: versioned::create(
+            VERSION,
+            squid_v0::new(
+                new_version_control(),
+                ctx,
+            ),
+            ctx,
+        ),
     });
 }
 
-public(package) fun borrow_channel(self: &Squid): &Channel {
-    &self.channel
+// ------
+// Macros
+// ------
+/// This macro also uses version control to sinplify things a bit.
+macro fun value($self: &Squid, $function_name: vector<u8>): &SquidV0 {
+    let squid = $self;
+    let value = squid.inner.load_value<SquidV0>();
+    value.version_control().check(VERSION, ascii::string($function_name));
+    value
+}
+
+/// This macro also uses version control to sinplify things a bit.
+macro fun value_mut($self: &mut Squid, $function_name: vector<u8>): &mut SquidV0 {
+    let squid = $self;
+    let value = squid.inner.load_value_mut<SquidV0>();
+    value.version_control().check(VERSION, ascii::string($function_name));
+    value
+}
+
+public(package) fun channel(self: &Squid, function_name: vector<u8>): &Channel {
+    self.value!(function_name).channel()
```

Comment:
```suggestion
    self.value!(b"channel").channel()
```

Reply:
this is a package function, so the public function that calls is needs to tell us what we need to check. It is used both by `transfers::its_transfer` and `discovery::register_transaction`

Reply:
Changed it but had to change the macros to use package functions instead.